lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -301,6 +301,71 @@ class AwsCompileFunctions {
         }
       }
 
+      const fileSystemConfigs = functionObject.fileSystemConfigs;
+
+      // TODO: Simplify these conditions
+      if (fileSystemConfigs && fileSystemConfigs.length !== 0) {
+        if (fileSystemConfigs.length > 1) {
+          // TODO: Add explanation - CloudFormation limitation
+          const errorMessage = 'You can specify at most 1 fileSystemConfig';
+          throw new this.serverless.classes.Error(errorMessage);
```

Comment:
Let's not put any validation here.

We've just added schema based config validation. So we should simply configure a JSON schema for those properties here: https://github.com/serverless/serverless/blob/d403d9c6d3dccdfc9f79d7c42b3cbc1a06fcfac0/lib/plugins/aws/provider/awsProvider.js#L145